tcwebhooks-core/src/main/java/webhook/teamcity/payload/content/WebHooksChange.java

```diff
@@ -5,15 +5,34 @@
 
 import jetbrains.buildServer.vcs.SVcsModification;
 import jetbrains.buildServer.vcs.VcsFileModification;
+import jetbrains.buildServer.vcs.VcsRootInstance;
+import org.jetbrains.annotations.Nullable;
 
 
 public class WebHooksChange {
 
+	private static @Nullable String tryGetVcsRootName(final SVcsModification modification) {
+		if(modification.isPersonal()) {
```

Comment:
Do we need this as well as the try/catch?

Would `modification.isPersonal()` ever return false, and then throw the `UnsupportedOperationException`?

Reply:
I also asked myself if both checks are necessary, but ultimately decided to use both for the simple reason that I don't know if there exists any other subtypes of `SVcsModification` that are not necessarily representative of personal changes, but that might also raise an exception. In such a case, the exception handler would catch those as well. 

I'm more than open to changing the logic to fit whatever you think is best. The helper method could even simply do away with the `isPersonal` check and simply use a big try-catch on `Exception` rather than `UnsupportedOperationException` if that's something you want.

Reply:
Yes. That's good logic. There's also a chance the classes could change in future versions on TeamCity.

I will merge the pr as it is against `master` (1.2)  and cherry pick it onto `1.1.x.x`. I'll try to get a 1.1 release out over Easter. 1.2 is a while away as we have a new baby.